src/main/java/ui/components/FilterTextField.java

```diff
@@ -48,13 +48,14 @@ public FilterTextField(String initialText) {
     private void setup() {
         setPrefColumnCount(30);
         validationSupport.registerValidator(this, (c, newValue) -> {
-            boolean wasError = false;
+            boolean isValidFilter = true;
             try {
                 Parser.parse(getText());
             } catch (ParseException e) {
```

Comment:
Is there a reason why we are catching ParseException and not catching FilterException (which in includes both ParseException and SemanticExpection)?